File: LeetCode/Greedy/MaximumNumberOfCoinsYouCanGet.java
// https://leetcode.com/problems/maximum-number-of-coins-you-can-get/description/?envType=problem-list-v2&envId=greedy


class Solution {
    public int maxCoins(int[] piles) {
        Arrays.sort(piles);
        int  sum= 0;
        for (int i = piles.length - 2 ;  i > (piles.length / 3) -1 ; i-=2) {
            sum += piles[i];
        }
        return sum ;
    }
}

File: LeetCode/Greedy/MaximumNumberOfCoinsYouCanGet.py
#  https://leetcode.com/problems/maximum-number-of-coins-you-can-get/description/

class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        piles.sort()
        n = (len(piles) // 3) * 2

        ans, i = 0, 1
        while i < n:
            ans += piles[-(i+1)]
            i += 2

        return ans